components/server/src/workspace/suggested-repos-sorter.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { SuggestedRepository } from "@gitpod/gitpod-protocol";
+
+const PRIORITY_LOW = 1;
+const PRIORITY_MEDIUM = 2;
+const PRIORITY_HIGH = 5;
+
+export type SuggestedRepositoryWithSorting = SuggestedRepository & {
+    priority: number;
+    lastUse?: string;
+};
+
+export const sortSuggestedRepositories = (repos: SuggestedRepositoryWithSorting[]) => {
+    // First we need to make a unique list and merge properties for entries with the same repo url
+    // This allows us to consider the lastUse of a recently used project when sorting
+    // as it will may have an entry for the project (no lastUse), and another for recent workspaces (w/ lastUse)
+    const uniqueRepositories = new Map<string, SuggestedRepositoryWithSorting>();
+    for (const repo of repos) {
+        const existingRepo = uniqueRepositories.get(repo.url);
+
+        const mergedEntry = {
+            ...(existingRepo || repo),
+            priority: existingRepo?.priority === undefined ? repo.priority : existingRepo.priority + repo.priority,
```

Comment:
This could be overkill, but it accomplished what I was shooting for, and feels like it could let us tweak how entries are ranked a bit more too.